libs/cardano-ledger-api/src/Cardano/Ledger/Api/Era.hs

```diff
@@ -400,15 +400,19 @@ upgradeTxDats ::
 upgradeTxDats (TxDats datMap) = TxDats $ fmap upgradeData datMap
 
 translateAlonzoTxAuxData ::
-  (AlonzoEraScript era1, AlonzoEraScript era2) =>
+  ( AlonzoEraScript era1
+  , AlonzoEraScript era2
+  , Coercible (NativeScript era1) (NativeScript era2)
+  ) =>
   AlonzoTxAuxData era1 ->
   AlonzoTxAuxData era2
-translateAlonzoTxAuxData AlonzoTxAuxData {atadMetadata, atadTimelock, atadPlutus} =
+translateAlonzoTxAuxData AlonzoTxAuxData {atadMetadata, atadNative, atadPlutus} =
   AlonzoTxAuxData
     { atadMetadata = atadMetadata
-    , atadTimelock = translateTimelock <$> atadTimelock
+    , atadNative = coerce atadNative
```

Comment:
You'd want to remove all uses of `translateAlonzoTxAuxData` function in this module
GHC will not notify you about this deprecation, if it is used in the same module in which it is defined in, so you'd need to do it your self.

However, once you try to remove it, you will see that it is actually useful in this module.

So, instead of deprecating it I would suggest to use the newly added function and continue using `translateAlonzoTxAuxData` in this module:

```suggestion
    , atadNativeScripts = upgradeNativeScript <$> atadNativeScripts
```

Reply:
> Should I then replace era2 with era and era1 with PreviousEra era, since upgradeNativeScript expects consecutive eras?

Yes, as we've talked on the call, this will be necessary